client/state/notices/account-recovery/index.js

```diff
@@ -6,9 +6,43 @@ import { translate } from 'i18n-calypso';
 /**
  * Internal dependencies
  */
-import { errorNotice } from 'state/notices/actions';
+import {
+	successNotice,
+	errorNotice,
+} from 'state/notices/actions';
+
+import {
+	getAccountRecoveryPhone,
+	getAccountRecoveryEmail,
+} from 'state/account-recovery/settings/selectors';
+
 import { dispatchError } from '../utils';
 
+const getUpdateSuccessMessage = ( target, getState ) => {
+	switch ( target ) {
+		case 'phone':
+			const oldPhone = getAccountRecoveryPhone( getState() );
+
+			if ( null == oldPhone ) {
+				return translate( 'Successfully added new recovery SMS number.' );
+			}
+
+			return translate( 'Successfully updated recovery SMS number.' );
+
+		case 'email':
+			const oldEmail = getAccountRecoveryEmail( getState() );
+
+			if ( '' === oldEmail ) {
```

Comment:
The line https://github.com/Automattic/wp-calypso/blob/master/client/state/account-recovery/settings/reducer.js#L90 should guarantee the value won't be null. While I can surely leave it null there, it's actually my personal preference to have an empty string to indicate the absence of a string field, and to have a null to indicate the absence of an object.

Does that make sense to you?